code/messaging/src/main/java/com/adobe/marketing/mobile/messaging/ContentCardSchemaDataUtils.kt

```diff
@@ -0,0 +1,213 @@
+/*
+  Copyright 2024 Adobe. All rights reserved.
+  This file is licensed to you under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License. You may obtain a copy
+  of the License at http://www.apache.org/licenses/LICENSE-2.0
+  Unless required by applicable law or agreed to in writing, software distributed under
+  the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+  OF ANY KIND, either express or implied. See the License for the specific language
+  governing permissions and limitations under the License.
+*/
+
+package com.adobe.marketing.mobile.messaging
+
+import com.adobe.marketing.mobile.aepcomposeui.AepUI
+import com.adobe.marketing.mobile.aepcomposeui.SmallImageUI
+import com.adobe.marketing.mobile.aepcomposeui.state.SmallImageCardUIState
+import com.adobe.marketing.mobile.aepcomposeui.uimodels.AepButton
+import com.adobe.marketing.mobile.aepcomposeui.uimodels.AepIcon
+import com.adobe.marketing.mobile.aepcomposeui.uimodels.AepImage
+import com.adobe.marketing.mobile.aepcomposeui.uimodels.AepText
+import com.adobe.marketing.mobile.aepcomposeui.uimodels.AepUITemplate
+import com.adobe.marketing.mobile.aepcomposeui.uimodels.AepUITemplateType
+import com.adobe.marketing.mobile.aepcomposeui.uimodels.SmallImageTemplate
+import com.adobe.marketing.mobile.services.Log
+import com.adobe.marketing.mobile.util.DataReader
+
+/**
+ * Utility object for creating instances of AEP model classes from Map data structures.
+ * This utility handles parsing and transforming content card data into the relevant AEP UI models,
+ * such as [AepText], [AepImage], [AepButton], and [AepDismissButton], as well as creating
+ * the corresponding UI templates like [SmallImageTemplate].
+ */
+internal object ContentCardSchemaDataUtils {
+    const val SELF_TAG = "ContentCardSchemaDataUtils"
+    /**
+     * Creates an instance of [AepText] from a map of key-value pairs.
+     *
+     * @param map The map containing the data for [AepText].
+     * @return An instance of [AepText].
+     */
+    private fun createAepText(map: Map<String, Any>): AepText {
+        return AepText(
+            content = DataReader.optString(map, MessagingConstants.ContentCard.UIKeys.CONTENT, "")
```

Comment:
nit: Can we also assign this to `null` for
1. Consistency with other methods
2. Causes exception if an AepText is created without `content` which should is erroneous and ensures template is null

Reply:
Done